---
title: "Survival Analysis of Breast Cancer Data from the TCGA Dataset"
author: "Ramaa Nathan"
date: "6/25/2019"
always_allow_html: yes
output:
  word_document:
    toc: yes
    toc_depth: '4'
  github_document:
    toc: true
    toc_depth: '4'
  pdf_document:
    df_print: kable
    number_sections: yes
    toc: yes
    toc_depth: '4'
  html_document:
    df_print: kable
    messages: no
    number_sections: yes
    toc: yes
    toc_depth: 4
    toc_float: yes
link-citations: yes
bibliography: SAReferences.bib
---


```{r setup, include=FALSE}
knitr::opts_chunk$set(echo = FALSE, cache=FALSE,tinytex.verbose = TRUE)

#The Following libraries need to be installed once.
# Load the bioconductor installer. 
# Try http:// if https:// doesn't work.
#source("https://bioconductor.org/biocLite.R")

# Install the main RTCGA package
#biocLite("RTCGA")

# Install the clinical and mRNA gene expression data packages
#biocLite("RTCGA.clinical")
#biocLite("RTCGA.mRNA")

# Test CRAN package installation:
library(tidyverse)
library(stringr) #for string manipulations
library(forcats) # for factors

library(knitr) # for kable
library(kableExtra) # for formatting tables

library(gridExtra) #for plotting in a grid

library(xtable) 
library(JM)
library(cmprsk)

#load survival libraries
library(survminer)
library(survival)

# Test RTCGA: 
library(RTCGA)
library(RTCGA.clinical)
library(RTCGA.mRNA)

#modeling
library(AICcmodavg)  #to extract AICc
library(parmsurvfit)
library(flexsurv)
library(SurvRegCensCov)

```

```{r,include=FALSE}
getOutputFormat <- function() {
  output <- rmarkdown:::parse_yaml_front_matter(
    readLines(knitr::current_input())
    )$output
  if (is.list(output)){
    return(names(output)[1])
  } else {
    return(output[1])
  }
}
```

# Introduction

Survival Analysis is a branch of statistics to study the  expected duration of time until one or more events occur, such as death in biological systems, failure in meachanical systems,  loan performance in economic systems, time to retirement, time to finding a job in  etc. In case of cancer studies, one of the primary objectives is to assess the time to an event of interest like relapse of cancer, death, etc.  

Survival Analysis is especially helpful in analyzing these studies when one or more of the cohorts do not experience the event and are considered censored for various reasons like death due to a different cause, loss-to-follow-up, end of study, etc.  The basic quantity used to describe time-to-event data is the survival function which is the probability of surviving beyond time x. 

The survival function can be modeled using parametric methods (Exponential, Weibull, etc), semi-parametric methods (Cox proportional hazards model), and non-parametric methods (Kaplan Meier model). The difference in survival times due to different treatment groups can also be compared using Logrank tests. The Cox proportional hazards model is useful in modeling the survival function in the presence of covariates. 

[The Cancer Genome Atlas (TCGA) Program](https://www.cancer.gov/about-nci/organization/ccg/research/structural-genomics/tcga), a joint effort between the National Cancer Institute and the Human Genome Research Institute, provides publicly-available clinical and high-throughput genomic data for thirty-three different types of cancers. This rich data source is widely used by researchers and has led to vast improvements in diagnosing, treating, and preventing cancer. The TCGA dataset will be used in this study to do survival analysis of breast cancer data.

# Research Questions
Here are a few questions that we could answer with this study.

1. What is the probability of survival for breast cancer?
2. How do different cancers (breast, ovarian, lung) affect survival rates?
3. What are the important factors that influence estimation of survival rate for Breast Cancer?
4. What are the effects of each factor on survival?  
5. What is the probability of survival for breast cancer when other clinical covariates are considered?

  
# Survival Analysis Basics
Before we start analyzing the data, lets first try to understand the basic terminologies related to survival analysis.

## Terminologies
### Event Times
Event times often are useful endpoints in clinical trials. Examples include survival time from onset of diagnosis, time until progression from one stage of disease to another, and time from surgery until hospital discharge. In each case, time is measured from study entry until the event occurs. 

### Censoring 
With an endpoint that is based on an event time, there always is the chance of censoring. An event time is censored if there is some amount of follow-up on a subject and the event is not observed during the study period. There are different types of censoring [@clinicalTrials]


### Survival Function
The survival function is the probability of surviving beyond time t or the probability of experiencing the event beyond time t. The survival function takes value 1 at the origin and 0 at infinity. <br>
$$ S(t) = P(T > t) $$
If f(t) is the probability density function (pdf) that describes the time-to-event, and F(t) is the corresponding cumulative distributive function, then <br>
<!--  $$ -->
\begin{align}
F(t) &= \int_{0}^{t} f(x)~dx \\
S(t) &= 1 - F(t) = \int_{t}^{\infty} f(x)~dx 
\end{align} 
<!--  $$ -->

### Hazard Rate or Hazard Function
The hazard function, h(x) is defined as the instantaneous risk of the event or  the probability that if a person survives to t, they will experience the event in the next instant. The hazard function or hazard rate can be considered to be the slope of the survival function.<br>

$$ h(t) = \lim_{\Delta t -> 0} \frac{P[t \le T < t+\Delta\,t | T \ge t]}{\Delta t} $$ 
The hazard function can also be expressed as the ratio of the probability density function to the survival function
$$  h(t) = \frac{f(t)}{S(t)} $$

### Cumulative hazard
Cumulative hazard is the accumulation of hazard rate over time. It is not a probability and is a measure of the risk. The greater the value of H(t), the greater the risk of failure by time t. When the distribution is continuous, the cumulative hazard is the integral of the hazard rate.
$$ H(t) = \int_0^t h(x) dx = -log( S(t) )$$

### Hazard Ratio
Hazard ratio is defined as the ratio of of two hazard functions, corresponding to two treatment groups

$$ \Lambda = \frac{h_1 (t)}{h_2 (t)} $$


### Proportional Hazard
The hazard ratio can be used to compare two treatment groups. When the hazard ratio is constant and independent of time, the hazards for the two treatment groups are said to be proportional. Basically, the relative risk of the event is constant over time. 

### Non-parametric Kaplan Meier Model 
The Kaplan Meier survival curve is a non-parametric teachnique for estimating the probability of survival, even in the presence of censoring. In this model, there is the notion of a risk set, which is the set of all individuals who are at risk to have an event at time t. This includes individuals who are known to be alive at time t and those who have the actual event at time t.  

In the modeling process, the actual failure times (event times) are first ordered in an increasing order. At each event time $t_k$, the number of subjects still at risk ($n_k$), the number of events ($d_k$), number of censored (lost to follow-up) subjects since the last event time ($n_k-d_k$) are recorded. The risk set does not include the subjects lost to follow-up. The Kaplan Meier Survival probability at time $t_k$ utilizes conditional probability - the probability of surviving at time t, given that the person has survived upto time t.
$$
S(t) =
\begin{cases}
1 & t_0 \le t \le t_1 \\
\prod_{i=1}^{i=k} (\frac{n_i - d_i}{n_i}) & t_k \le t \le t_{k+1}, k=1,2,...,K
\end{cases}
$$

Tha basic assumptions in a Kaplan Meier model are:

    1. The censoring is independent of prognosis
    2. The survival probabilities are the same for all subjects recruited at any time in the study
    3. Events happened at the time specified.  

### Parametric Exponential Distribution
<!-- $$ -->
\begin{align}
pdf: f(t) &= \lambda e^{-\lambda t} \\
cdf: F(t) &= 1 - e^{-\lambda t} \\
Survival \ Function: S(t) &= 1 - F(t) = e^{-\lambda t} \\
Hazard \ Function: h(t) &= \frac{f(t)}{S(t)} = \lambda \\
Hazard \ Ratio = \frac{\lambda_1}{\lambda_2} \\
Cumulative \ Hazard: H(t) &= -log S(t) = \lambda t
\end{align}
<!-- $$ -->
In the case of an exponential distribution, the hazard function or the hazard rate is a constant $\lambda$ and the hazard ratio is proportional as it is independent of time. 


### Semi-Parametric Cox Proportional Hazard Regression Model
The Cox Regression Model is used to model the hazard at time t in the presence of multiple covariates, each of which could be categorical or quantitative. The Cox model is similar to the exponential model where the survival time is given by $S(t) = e^{\lambda t)}$. But the hazard rate  $\lambda$  is now considered to be a linear combination of several covariates $Z=Z_1, Z_2, ..,Z_p$ and so $$\lambda(Z_1, Z_2, ..,Z_p) = \beta_1Z_1 + \beta_2Z_2 +..+ \beta_pZ_p$$
The Cox regression model can then be expressed as 
$$ h(t|Z) = h_0(t)exp(\Sigma_{k=1}^{p}\beta_kZ_k)$$ where $h(t|Z)$ is the hazard at time t for an individual with covariates Z and $h_0(t) = h(t|Z=0)$ is the baseline hazard rate.  

The Cox model is semi-parametric, containing  both parametric and non-parametric components. 

1. The $h_0(t)$ is the non-parametric component and can take any form as along as $h_0(t) \ge 0$ 

2. $exp(\Sigma_{k=1}^{p}\beta_kZ_k)$ is the parametric component.

For example, suppose we have three predictors - therapy, age and race. Then
$$ h(t) = h_0(t) exp(\beta_1 therapy + \beta_2age + \beta_3 race) $$

So, the coefficient $\beta_1$ is the log of the hazard ratio and $exp(\beta_1)$ is the hazard ratio for the individual on treatment B compared to treatment A, when the race and age covariates are the same for both individuals. The values of $exp(\beta1)$ provide the following interpretations,

* $exp(\beta_1)$ > 1 indicates higher hazard or lower survival rate compared to the base hazard function.
* $exp(\beta_1)$ < 1 indicates lower hazard or higher survival rate 
* $exp(\beta_1)$ = 1 indicates no association 
   
 In R, we use the coxph function to model the Cox PH model. It provides the following output values:
  
1. coef = the estimate of $\beta_i$
2. exp(coef) - the estimate of $exp(\beta_i)$ 
3. se(coef) - the standard error of the estimate of $\beta_i$
4. z = $\frac{z}{se(coef)}$ = the Wald statistic for testing the null hypothesis that $\beta_i = 0$ assuming that z follows a standard normal distribution
5. p = two sided p-value

One of the basic assumptions of the Cox Proportional Hazards Model is that the hazards are proportional. This can be tested by checking that the Schoenfeld residuals exhibit a random pattern against time. In R, cox.zph function in the survival package and ggcoxzph from the survminer package can be used to check for the plot of these residuals. 


# R Functions for Survival Analysis
The survival analysis in this study has been done in R and here is a list of the important functions used in the analysis.
```{r, echo=FALSE}
rfns_table <- tribble(
 ~"Purpose", ~"Package::Function", ~"Package::Graphical Wrapper",
 "Extract Survival Data from TCGA","rtcga::survivalTCGA"," ",
 "Create Survival Object", "survival::Surv()", " ",
 "Fit a Kaplan Meier Curve","survival::survfit","survminer::ggsurvplot()",
 "Compare Kaplan Meier Curves using logrank","survival::survdiff()", "survminer::ggsurvplot()",
 "Fit a parametric model", "survival::survreg", " ",
 "Fit Cox Proportional Hazards Model","survival::coxph()","survminer::ggforest()",
 "Test for Proportional Hazards","survival::cox.zph()","survminer::ggcoxzph()",
 "Display Adjusted Survival Curves for Cox Proportional Hazards Model for a Factor"," ","survminer::ggcoxadjustedcurves",
 "Split the survival data set at specific cut times to accommodate the time-dependent covaraites for Cox","survival::survSplit"," ",
 "Convert Weibull results to an easy interpretable form","SurvRegCensCov::ConvertWeibull"," "
)
#rfns_table %>% kable() %>%
#    kable_styling(full_width=T,bootstrap_options=c("striped","bordered")) %>%
#  add_header_above(c("R Functions for Survival Analysis" = 3))
if(getOutputFormat() == 'pdf_document') {
   rfns_table %>% kable("latex",booktabs=T) %>%
    kable_styling(full_width=T,position="center",#font_size=9, 
                  bootstrap_options=c("bordered","striped","responsive","condensed")) %>%
    column_spec(1,width="20em",border_left=T,border_right=T) %>%
    column_spec(2,width="10em",border_right=T) %>%
    column_spec(3,width="20em",border_right=T) %>%
    row_spec(row=0,bold=TRUE) %>%
    add_header_above(c("R Functions for Survival Analysis" = 3))
} else {
  rfns_table %>% kable() %>%
    kable_styling(full_width=T,position="center",#font_size=9, 
                  bootstrap_options=c("bordered","striped","responsive","condensed")) %>%
    column_spec(1,width="20em",border_left=T,border_right=T) %>%
    column_spec(2,width="10em",border_right=T) %>%
    column_spec(3,width="20em",border_right=T) %>%
    row_spec(row=0,bold=TRUE) %>%
   add_header_above(c("R Functions for Survival Analysis" = 3))
}
``` 

# Data Source and Description
 The Cancer Genome Atlas (TCGA) Program [1] provides publicly-available clinical and high-throughput genomic data for thirty-three different types of cancers. For this study of survival analysis of Breast Cancer, we use the Breast Cancer (BRCA) clinical data that is readily available as BRCA.clinical. This dataset has 3703 columns from which we pick the following columns containing demographic and cancer stage information as important predictors of survival analysis.
 
```{r origcolumns, echo=FALSE}
clinicalDataCols = tibble(
  ColumnName = c("Gender", "Race", "Ethnicity","Age","Vital Status",
             "Days to Death","Days to Followup",
             "Therapy type",
             "Pathologic Stage",
             "Pathology T",
             "Pathology N",
             "Pathology M"
             ),
  DataType = c("categorical","categorical","categorical","integer","binary","integer",
               "integer","categorical","categorical","categorical","categorical","categorical"),
  Description = c("Gender", "Race", "Ethnicity", "Age at first diagnosis", "Vital Status (1 - dead (event), 0 - alive/censored)",
                  "Number of days to death from first diagnosis",
                  "Number of days to last follow-up from first diagnosis",
                  "Therapy Type (Chemo, Harmone, Immuno, etc.",
                  "Cancer stage - based on T,M, and N labeling",
                  "Tumor (T) stage describing size and location of tumor",
                  "Lymph (N) nodes status describing if cancer has spread into nearby lymph nodes",
                  "Metastasis (M) status describing if cancer has spread to other parts of the body")
  )

if(getOutputFormat() == 'pdf_document') {
  clinicalDataCols %>%  kable("latex",booktabs=T) %>% 
    kable_styling(full_width=T,
                  bootstrap_options=c("bordered","striped","responsive","condensed")) %>%
    column_spec(column=1,border_left=TRUE,border_right=TRUE) %>%
    column_spec(column=2,border_right=TRUE) %>%
    column_spec(column=3,border_right=TRUE) %>%
    row_spec(row=0,bold=TRUE)
} else {
  clinicalDataCols %>%  kable() %>% 
    kable_styling(full_width=T,
                  bootstrap_options=c("bordered","striped","responsive","condensed")) %>%
    column_spec(column=1,border_left=TRUE,border_right=TRUE) %>%
    column_spec(column=2,border_right=TRUE) %>%
    column_spec(column=3,border_right=TRUE) %>%
    row_spec(row=0,bold=TRUE)
}

```



# Exploratory Data Analysis
The clinical data set from the The Cancer Genome Atlas (TCGA) Program is a snapshot of the data from 2015-11-01 and is used here for studying survival analysis.

## Data Extraction
The RTCGA package in R is used for extracting the clinical data for the Breast Invasive Carcinoma Clinical Data (BRCA). In addition, the survival and survminer packages in R are used for the analysis.

The survivalTCGA function in the RTCGA package is used to extract the relevant columns.  This function also uses the vital status variable that indicates if the observation was an event or a censor and  combines the number of days to death from first diagnosis and number of days to last follow-up from first diagnosis into a new "times" variable. 

```{r clinical, echo=FALSE}
# Create the clinical data
clin <- survivalTCGA(BRCA.clinical, OV.clinical, GBM.clinical, 
                     extract.cols=c("admin.disease_code","patient.drugs.drug.therapy_types.therapy_type"))

```

```{r BRCA_extract, echo=FALSE}
# Create the clinical data
brca_clin_orig <- survivalTCGA(BRCA.clinical, 
                     extract.cols=c("patient.gender", "patient.race",
                                    "patient.ethnicity","patient.days_to_birth","patient.vital_status",
                                    "patient.drugs.drug.therapy_types.therapy_type",
                                    "patient.stage_event.pathologic_stage",
                                    "patient.stage_event.tnm_categories.pathologic_categories.pathologic_t",
                                    "patient.stage_event.tnm_categories.pathologic_categories.pathologic_n",
                                    "patient.stage_event.tnm_categories.pathologic_categories.pathologic_m"))
#sapply(brca_clin_orig,class)
```

## Checks for Missing and Invalid Data
The extracted data is first checked for any missing data or invalid data. There are two observations that have negative values for the time to event that are filtered out during the data cleanup and transformation step.

```{r checkdata, include=FALSE}
# Check for number of missing data in each column
brca_clin_orig %>% sapply(function(x) sum(is.na(x)))

#Check number of rows with missing data
na_rows <- brca_clin_orig %>% apply(MARGIN=1, FUN=function(x) sum(is.na(x))) 
sum(na_rows>0)

brca_clin_orig %>% sapply(summary)
brca_clin_orig %>% filter(times < 0) %>% nrow()
```

The data is generally clean with only some of the demographic information like race and ethnicity missing for a few of the observations.  We also find that more than 40% of the rows have NAs in one or more columns. So, we will filter out the missing data, as needed during the analysis.

## Data Cleaning and Transformation
Next, the following cleaning and transformations are applied to the TCCGA BRCA.clinical data to get a clean and compact dataset:

1. Rename the long variable names to short names.
2. Filter out the 12 observations corresponding to males diagnosed with breast cancer.
3. Filter out the 2 observations with negative "times" value.
4. Create a "age" variable that contains the number of days at first diagnosis to age in years at first diagnosis.
5. Create a "years_to_event" variable which is the "times"" variable   converted from days to years.
6. Data in the pathology columns contain information on both stage and sub-stage. Transform the data to only contain the high level stage information.
5. Modify the "therapy_type"" to contain three types - chemotherapy, hormone therapy and Other (lump all the other infrequent types into Other)
6. Modify the "race"" to contain three types - black or african american and white (lump the other two types into Other)

```{r BRCA_transform, echo=FALSE}
#helper Functions
clean_pathologic_stage <- function(x) {
  x %>% str_replace_all(c(
    "stage iv[a-d]*"="stage4",
    "stage [i]{3}[a-d]*"="stage3",
    "stage i{2}[a-d]*"="stage2",
    "stage i{1}[a-d]*"="stage1",
    "stage x"="stageX"))
}

clean_pathologyTstage <- function(x) {
  x %>% str_replace_all(c(
    "\\s*tx"="tx",
    "\\s*t1[a-z]*"="t1", 
    "\\s*t2[a-z]*"="t2",
    "\\s*t3[a-z]*"="t3",
    "\\s*t4[a-z]*"="t4"))
}

clean_pathologyMstage <- function(x) {
  x %>% str_replace_all(c("cm0\\s+\\(i[\\+,-]\\)"="cm0"))
}

clean_pathologyNstage <- function(x) {
  x %>% str_trim(.) %>% 
    str_replace_all(c(
      "nx"="nx", 
      "n1[a-z]*"="n1", 
      "n2[a-z]*"="n2",
      "n3[a-z]*"="n3",
      "n4[a-z]*"="n4",
      "n0"="n0",
      "n0\\s+\\([a-z]*[\\+|-]\\)"="n0"))
}


#transform the data
brca_clin <- brca_clin_orig %>% 
  rename(gender=patient.gender,
         race=patient.race,
         ethnicity=patient.ethnicity,
         vital_status=patient.vital_status,
         therapy_type=patient.drugs.drug.therapy_types.therapy_type,
         pathologic_stage=patient.stage_event.pathologic_stage,
         pathologyTstage=patient.stage_event.tnm_categories.pathologic_categories.pathologic_t,
         pathologyNstage=patient.stage_event.tnm_categories.pathologic_categories.pathologic_n,
         pathologyMstage=patient.stage_event.tnm_categories.pathologic_categories.pathologic_m) %>%
  filter(gender == "female") %>% 
  filter(times > 0) %>%
  mutate(age=abs(as.numeric(patient.days_to_birth))/365,
         therapy_type = ifelse(is.na(therapy_type),"No Info",therapy_type),
         therapy_type = fct_lump(therapy_type,3),
         race = fct_lump(race,2),
         pathologic_stage=str_trim(pathologic_stage),
         pathologyTstage=str_trim(pathologyTstage),
         pathologyNstage=str_trim(pathologyNstage),
         pathologyMstage=str_trim(pathologyMstage),
         pathologic_stage = clean_pathologic_stage(pathologic_stage),
         pathologyTstage = clean_pathologyTstage(pathologyTstage),
         pathologyNstage = clean_pathologyNstage(pathologyNstage),
         pathologyMstage = clean_pathologyMstage(pathologyMstage),
         years_to_event=times/365,
         agecat=cut(age, breaks=c(0, 40, 60, Inf), labels=c("young", "middle", "old"))
         )

#Convert specified columns from character to factor type.
convert_to_factor <- c("ethnicity", "pathologic_stage",
                       "pathologyTstage", "pathologyMstage","pathologyNstage")
brca_clin <- brca_clin %>% mutate_at(convert_to_factor,factor)

#remove unnecessary columns
brca_clin <- brca_clin %>% 
  dplyr::select(-starts_with("patient"))
#names(brca_clin)

#Verify the class types
#sapply(brca_clin,class)

#What are the dimensions of the brca_clin dataset?
#dim(brca_clin)
```

## Data Visualizations


### Histograms
**Age at First Diagnosis**
```{r}
brca_clin %>% filter(!is.na(age)) %>% 
  ggplot(aes(x=age)) +
    geom_histogram(aes(y=..density..),color="black",fill="white")+
    geom_density(alpha=0.2,fill="#FF6666") +
    geom_vline(aes(xintercept=mean(age)), color="blue", linetype="dashed", size=1) +
    labs(title="Age at First Diagnosis")

brca_clin %>% filter(!is.na(age)) %>% 
  ggplot(aes(x=age)) +
    geom_histogram(color="black",fill="white")+
    #geom_density(alpha=0.2,fill="#FF6666") +
    geom_vline(aes(xintercept=mean(age)), color="blue", linetype="dashed", size=1) +
    labs(title="Age at First Diagnosis")
    
```

The average age at first diagnosis seems to 59 years and the distribution of age is mostly symmetrical with a  small bimodal effect. 

**Time to Event**  
```{r}
brca_clin %>% filter(!is.na(times)) %>%
  ggplot(aes(x=years_to_event)) +
    geom_histogram(color="black",fill="white") +
    geom_vline(aes(xintercept=mean(years_to_event)), color="blue", linetype="dashed", size=1) +
    labs(title="Time to Event")

brca_clin %>% dplyr::select(age) %>% summary()
```

Without considering censoring the mean survival time seems to be less than 2.5 years. This is not helpful information as censoring information is an important component of survival data.  The TCGA clinical data set is a survival dataset containing  right censor information. We will first visualize the distribution of the right censor data by different categories.  The Censoring Event plots here were inspired by a [workshop on Survival Analysis](https://github.com/emsweene/New_York_R_Survival_Analysis_Workshop).


```{r, echo=FALSE}
#use age as a continuous variable
anova(lm(age ~ therapy_type, data=brca_clin))
```


### Censoring and Event Plots  by Disease

```{r disease_plot, echo=FALSE}
#Contingency table
#xtabs(~admin.disease_code+patient.vital_status,data=clin) %>% addmargins()
byDisease <- xtabs(~admin.disease_code+patient.vital_status,data=clin) %>% addmargins()
#names(byDisease) <- c("Disease", "Vital Status", "Total")
byDisease %>% knitr::kable() %>%
  kable_styling(full_width=F,position="float_left",bootstrap_options=c("bordered")) 

clin <- clin %>%
  mutate(years_to_event=times/365) %>%
  filter(!is.na(admin.disease_code)) %>%
  arrange(admin.disease_code) 

clin %>%
  mutate(index=1:n()) %>% 
  ggplot(
       aes(xend = 0, 
           y = index, 
           x = years_to_event, 
           yend = index, 
           colour = admin.disease_code,
           shape = factor(patient.vital_status))) + 
  geom_segment() + 
  geom_point() +
  ggtitle("Right Censoring in TCGA - By Disease") +
  labs(x="Years to Event", y="Subjects") +
  scale_shape_discrete(name = "Status", labels = c("Censored","Event")) +
  scale_color_discrete(name = "Disease", labels = c("BRCA (Breast Cancer)", "GBM (Glioblastoma Multiforme)", "OV (Ovarian Cancer)"))


```

From the contingency table and the plots comparing the diseases of BRCA (breast cancer), GBM (Glioblastoma Multiforme) and OV (Ovarian Cancer), it can be observed that less that almost 50% of the cases are for Breast Cancer and the rest are almost equally split between ovarian and GBM. In these, 10% of the subjects with Breast cancer, 75% of the subjects with GBM and 50% of the subjects with ovarian cancer did not survive (had events). 


### Censoring and Event Plots  by Age Category

```{r age_plot, echo=FALSE}
#Contingency table
xtabs(~agecat+vital_status,data=brca_clin) %>% addmargins()

brca_clin_age <- brca_clin %>%
  filter(!is.na(agecat)) %>%
  arrange(agecat) 

brca_clin_age %>%
  mutate(index=1:n()) %>% 
  ggplot(
       aes(xend = 0, 
           y = index, 
           x = years_to_event, #times, 
           yend = index, 
           colour = agecat,
           shape = factor(vital_status))) + 
  geom_segment() + 
  geom_point() +
  ggtitle("Right Censoring in TCGA - BRCA by Age Categories") +
  labs(x="Years to Event", y="Subjects") +
  scale_shape_discrete(name = "Status", labels = c("Censored","Event")) +
  scale_color_discrete(name = "Age Categories", labels = c("Young", "Middle", "Old"))

```

From the contingency table and the plots, it can be observed that less than 10% of the subjects are less than 40 years old. The number of subjects in the other two groups are almost equally distributed of which more number of events were observed in the older age group. 



### Censoring and Event Plots  by Pathological Stage
```{r stage_plot, echo=FALSE}
#Contingency table
xtabs(~pathologic_stage+vital_status,data=brca_clin) %>% addmargins()

brca_clin_stage <- brca_clin %>%
  filter (!is.na(pathologic_stage)) %>%
  arrange(pathologic_stage)

brca_clin_stage %>%
  mutate(index=1:n()) %>% 
  ggplot(
       aes(xend = 0, 
           y = index, 
           x = years_to_event, #times, 
           yend = index, 
           colour = pathologic_stage,
           shape = factor(vital_status))) + 
  geom_segment() + 
  geom_point() +
  ggtitle("Right Censoring in TCGA - BRCA by Pathologic (Cancer) Stage") +
  labs(x="Years to Event", y="Subjects") +
  scale_shape_discrete(name = "Status", labels = c("Censored","Event")) +
  scale_color_discrete(name = "Cancer Stage", labels = c("Stage1","Stage2","Stage3","Stage4","StageX"))


```

From the contingency table and the censor plots, it can be observed that almost 50% of the subjects are diagnosed with stage 2 cancer, with less than 2% of the subjects are either in stage 4 or in stage X. The percentage of events is the highest (almost 50%) for the stage 4  and stage X groups and the lowest for stage 1.



# Answers to Research Questions:
1. What is the median survival time for breast cancer?
   - In the absence of any predictors, the median survival time is 9.5 years.
2. How do different cancers (breast, ovarian, lung) affect survival rates?
   - Of the three types of cancers  BRCA (breast cancer), GBM (Glioblastoma Multiforme) and OV (Ovarian Cancer) that were compared, BRCA has the highest survival rate.
3. What are the important factors that influence risk of death for Breast Cancer?
   - When considered individually or together age at first diagnosis, no therapy, cancer stage 3 and cancer stage 4, and any metastasis into the lymph nodes significantly reduce survival time. 
4. What are the effects of each factor on survival?  
   - The number of lymph nodes that have been metasized into seems to be the strongest predictor of survival. 
5. What is the probability of survival for breast cancer when other clinical covariates are considered?
   - The results do not change when the predictors are all accounted for indicating an additive model with no interactions. 

---
nocite: |
  @clinicalTrials, @psu_notes, @SA_Ref1, @SA_Ref2, @SA_Ref3, @RSA_1, @TCGA1
---

# References